tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -5953,3 +5981,859 @@ def get_prod(arr):
+    # the result of the last operation is a Python complex type which has 8 bytes in the alignment,
+    # that's why we need to convert it to a numpy.complex64 the needed type.
+    cmplx64 = np.complex64(cmplx128_from_float32)
+    cmplx128 = uniform(low=-((abs(min_int) / size[0] * 2)**(1/2)), high=(max_int / size[0] * 2)**(1/2), size=size) + uniform(low=-((abs(min_int) / size[0] * 2)**(1/2)), high=(max_int / size[0] * 2)**(1/2), size=size) * 1j
+
+    f_bl = epyccel(get_matmul, language=language)
+
+    assert (f_bl(bl) == get_matmul(bl))
+
+    f_integer8 = epyccel(get_matmul, language=language)
+    f_integer16 = epyccel(get_matmul, language=language)
+    f_integer = epyccel(get_matmul, language=language)
+    f_integer32 = epyccel(get_matmul, language=language)
+
+    assert (f_integer8(integer8) == get_matmul(integer8))
+    assert (f_integer16(integer16) == get_matmul(integer16))
+    assert (f_integer(integer) == get_matmul(integer))
+    assert (f_integer32(integer32) == get_matmul(integer32))
+
+    # the if block should be removed after resolving (https://github.com/pyccel/pyccel/issues/735).
+    if sys.platform != 'win32':
+        f_integer64 = epyccel(get_matmul, language=language)
+        assert (f_integer64(integer64) == get_matmul(integer64))
+
+    f_fl = epyccel(get_matmul, language=language)
+    f_fl32 = epyccel(get_matmul, language=language)
+    f_fl64 = epyccel(get_matmul, language=language)
+
+    assert np.isclose(f_fl(fl), get_matmul(fl), rtol=RTOL, atol=ATOL).any()
+    assert np.isclose(f_fl32(fl32), get_matmul(fl32), rtol=RTOL32, atol=ATOL32).any()
+    assert np.isclose(f_fl64(fl64), get_matmul(fl64), rtol=RTOL, atol=ATOL).any()
+
+    f_complex64 = epyccel(get_matmul, language=language)
+    f_complex128 = epyccel(get_matmul, language=language)
+
+    assert np.isclose(f_complex64(cmplx64), get_matmul(cmplx64), rtol=RTOL32, atol=ATOL32).any()
+    assert np.isclose(f_complex128(cmplx128), get_matmul(cmplx128), rtol=RTOL, atol=ATOL).any()
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [pytest.mark.fortran,
+            pytest.mark.skip(reason="Still under maintenance, See #770")]),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="Needs a C printer see https://github.com/pyccel/pyccel/issues/791"),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = [
+            pytest.mark.python,
+            pytest.mark.skip(reason="Missing Python printer")]
+        )
+    )
+)
+
+# Not all arguments are supported
+
+def test_numpy_where_scalar(language):
+
+    @types('bool')
+    @types('int')
+    @types('int8')
+    @types('int16')
+    @types('int32')
+    @types('int64')
+    @types('float')
+    @types('float32')
+    @types('float64')
+    @types('complex64')
+    @types('complex128')
+    def get_where(a):
+        from numpy import where, shape
+        b = where(a)
```

Comment:
I think this is the only call to `where` that is currently acceptable. Is that correct?

But I wonder if it should be disallowed in pyccel. It is a rather complicated function which returns a tuple of arrays:
```
In [1]: import numpy as np

In [2]: x = np.ones((3,4,5))

In [3]: np.where(x)
Out[3]: 
(array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1,
        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2,
        2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]),
 array([0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 0, 0,
        0, 0, 0, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 0, 0, 0, 0,
        0, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3]),
 array([0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 0, 1,
        2, 3, 4, 0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 0, 1, 2, 3,
        4, 0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 0, 1, 2, 3, 4]))
```

I suspect that currently we don't handle it like this at all.

[Numpy's documentation](https://numpy.org/doc/stable/reference/generated/numpy.where.html) specifically advises against using it like this:
>When only condition is provided, this function is a shorthand for np.asarray(condition).nonzero(). Using nonzero directly should be preferred, as it behaves correctly for subclasses. The rest of this documentation covers only the case where all three arguments are provided.

@yguclu do you have an opinion about this?